### PR TITLE
Add LogPollingThread class for watching log files

### PR DESCRIPTION
**Pull request by sommern**

### CODE REVIEW

**Comment by sommern** [comment]
Python's logging module is thread-safe between python threads from the same process, but it is *not* safe for logging when multiple processes will be accessing the same files.

Using `>>` for appending text from the command line is safe when the text to append is <= 512 bytes, so these custom loggers should be ok as long as we keep our logging messages short.

File: git-keeper-core/gkeepcore/log_polling.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    def _start_watching_log_file(self, log_file: LogFileReader):
+        # Start watching the file at file_path. Do not call directly, pass
+        # file paths in through the queue
+
+        try:
+            byte_count = log_file.get_byte_count()
+            self._log_byte_counts[log_file] = byte_count
+        except LogFileException:
+            # FIXME - log this
```
Comment (bjcoleman) [inline git-keeper-core/gkeepcore/log_polling.py]: When will this be done?

File: git-keeper-core/gkeepcore/log_polling.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+        lines = []
+
+        try:
+            current_byte_count = log_file.get_byte_count()
+            old_byte_count = self._log_byte_counts[log_file]
+
+            # if the file shrunk, something went wrong
+            if current_byte_count < old_byte_count:
+                # FIXME - log this
```
Comment (bjcoleman) [inline git-keeper-core/gkeepcore/log_polling.py]: When will this be done?

File: git-keeper-core/gkeepcore/log_polling.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                # get any new data past the old byte count
+                new_data = log_file.get_data(old_byte_count)
+
+                self._log_byte_counts[log_file] = current_byte_count
+
+                # strip the data since split does not remove the trailing
+                # newline
+                lines = new_data.strip().split('\n')
+        except LogFileException as e:
+            # FIXME - log this
```
Comment (bjcoleman) [inline git-keeper-core/gkeepcore/log_polling.py]: When will this be done?

File: git-keeper-core/gkeepcore/log_polling.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                                              line))
+
+        # consume all new log files until the queue is empty
+        try:
+            while True:
+                new_log_file = self._add_log_queue.get(block=False)
+                if isinstance(new_log_file, LogFileReader):
+                    self._start_watching_log_file(new_log_file)
+                else:
+                    # FIXME - log this
```
Comment (bjcoleman) [inline git-keeper-core/gkeepcore/log_polling.py]: When will this be done?

**Comment by bjcoleman** [comment]
This all looks good - except those FIXMEs.

I scanned through them rather than read them for detail. 

One other thought:  Is there a way to test this?

**Pull request merged**